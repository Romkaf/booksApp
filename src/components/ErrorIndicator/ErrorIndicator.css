.root {
	position: fixed;
	top: 0;
	left: 0;
	right: 0;
	z-index: 99;
	display: flex;
	justify-content: center;
	padding: 25px 15px;
	color: var(--brown);
	background-color: var(--light-beige);
	border-color: var(--beige);
	animation: show 7s linear forwards;
}

.root span {
	font-size: 1.2rem;
}

@keyframes show {
	0% {
		transform: translateY(-100%);
	}
	20% {
		transform: translateY(0%);
	}
	80% {
		transform: translateY(0%);
	}
	100% {
		transform: translateY(-100%);
	}
}
