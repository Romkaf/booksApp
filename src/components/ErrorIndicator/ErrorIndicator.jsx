import React from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorIndicator.css';

const ErrorIndicator = ({ text }) => (
	<div className={styles.root}>
		<span>Error: {text}</span>
	</div>
);

ErrorIndicator.propTypes = {
	text: PropTypes.string,
};

export default ErrorIndicator;
