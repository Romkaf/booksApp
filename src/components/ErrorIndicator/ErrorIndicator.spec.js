import React from 'react';
import ErrorIndicator from './ErrorIndicator';
import { shallow } from 'enzyme';

describe('ErrorIndicator', () => {
	it('should render component', () => {
		const component = shallow(<ErrorIndicator />);
		expect(component).toMatchSnapshot();
	});
});
