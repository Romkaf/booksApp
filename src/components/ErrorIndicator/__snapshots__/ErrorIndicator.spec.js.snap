// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorIndicator should render component 1`] = `
<div
  className="root"
>
  <span>
    Error: 
  </span>
</div>
`;
